src/Contexts/ItemSearchContext.php

```diff
@@ -13,33 +14,56 @@ class ItemSearchContext extends CategoryContext implements ContextInterface
 
     public $isSearch;
     public $searchString;
+    public $suggestionString;
 
     public function init($params)
     {
         parent::init($params);
 
         $itemListOptions = [
-            'page'          => $this->getParam( 'page', 1 ),
-            'itemsPerPage'  => $this->getParam( 'itemsPerPage', '' ),
-            'sorting'       => $this->getParam( 'sorting', '' ),
-            'facets'        => $this->getParam( 'facets', '' ),
-            'query'         => $this->getParam( 'query', '' ),
-            'priceMin'      => $this->request->get('priceMin', 0),
-            'priceMax'      => $this->request->get('priceMax', 0)
+            'page'         => $this->getParam('page', 1),
+            'itemsPerPage' => $this->getParam('itemsPerPage', ''),
+            'sorting'      => $this->getParam('sorting', ''),
+            'facets'       => $this->getParam('facets', ''),
+            'query'        => $this->getParam('query', ''),
+            'priceMin'     => $this->request->get('priceMin', 0),
+            'priceMax'     => $this->request->get('priceMax', 0)
         ];
 
         $itemListOptions = SearchOptions::validateItemListOptions($itemListOptions, SearchOptions::SCOPE_SEARCH);
 
+        $this->isSearch = true;
+        $this->searchString = $itemListOptions['query'];
+
         $this->initItemList(
             [
-                'itemList' => SearchItems::getSearchFactory( $itemListOptions ),
-                'facets'   => Facets::getSearchFactory( $itemListOptions )
+                'itemList' => SearchItems::getSearchFactory($itemListOptions),
+                'facets'   => Facets::getSearchFactory($itemListOptions)
             ],
             $itemListOptions,
             SearchOptions::SCOPE_SEARCH
         );
 
-        $this->isSearch = true;
-        $this->searchString = $itemListOptions['query'];
+        //try to get results for the did you mean search
+        if ((int)$this->itemCountTotal === 0 && !is_null($this->itemListAdditional)) {
+            /** @var ItemSearchAutocompleteService $itemSearchAutocompleteService */
+            $itemSearchAutocompleteService = pluginApp(ItemSearchAutocompleteService::class);
+            $itemListOptions['query'] = $itemSearchAutocompleteService->getDidYouMeanSuggestionSearchString(
+                $this->searchString,
+                $this->itemListAdditional['suggestions']
+            );
+
+            if (strlen($itemListOptions['query']) && $itemListOptions['query'] !== $this->searchString) {
+                $this->suggestionString = $itemListOptions['query'];
+                $this->initItemList(
+                    [
+                        'itemList' => SearchItems::getSearchFactory($itemListOptions),
+                        'facets'   => Facets::getSearchFactory($itemListOptions)
+                    ],
+                    $itemListOptions,
+                    SearchOptions::SCOPE_SEARCH
+                );
+            }
+        }
```

Comment:
Ich hätte den Block ebenfalls in den `ItemListContext` gepackt. So sparst du dir die Variable `$itemListAdditional` komplett und kannst direkt auf die Suggestions der ersten Suche zugreifen. Außerdem wird in der Klasse noch geprüft, ob es eine externe Suche gibt. Die "Meinten Sie"-Suche sollte nur ausgeführt werden, wenn die interne Suche keine Ergebnisse liefert.